src/test/java/usecasetesting/ReplyUseCaseTest.java

```diff
@@ -0,0 +1,162 @@
+        postInputBoundary.createPost(inputData2);
+        Question question = questionGateway.getQuestion(QUESTION_ID);
+        Post post2 = question.getPosts().get(0);
+        assertEquals(post2.getBelongsTo(), ATTORNEY_ID);
+        assertEquals(post2.getPostText(), "Test text");
+        User user = attorneyGateway.getUser(ATTORNEY_ID);
+        Question attorneyQuestion = user.getQuestionsList().get(0);// AttorneyQuestionList is not updated
+        assertEquals(attorneyQuestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyQuestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyQuestion.isTaken(), true);
+    }
+    @Test
+    public void testAttorneyFollowUp(){
+        setUpReplyUseCase();
+        PostRequestModel inputData = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData);
+
+        User user = attorneyGateway.getUser(ATTORNEY_ID);
+        Question attorneyquestion = user.getQuestionsList().get(0);// AttorneyQuestionList is not updated
+        assertEquals(attorneyquestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyquestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyquestion.isTaken(), true);
+
+        PostRequestModel inputData2 = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text2");// Attorney
+        postInputBoundary.createPost(inputData2);
+        Question question = questionGateway.getQuestion(QUESTION_ID);
+        Post post2 = question.getPosts().get(1);
+        assertEquals(post2.getBelongsTo(), ATTORNEY_ID);
+        assertEquals(post2.getPostText(), "Test text2");
+        assertEquals(attorneyquestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyquestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyquestion.isTaken(), true);
+
+    }
+    @Test
+    public void testFailToReplyQuestionClosed(){
+        setUpReplyUseCase();
+        PostRequestModel inputData = new PostRequestModel(CLOSED_QUESTION_ID, ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData);
+    }
+
+    @Test
+    public void testAttorneyFailToReplyQuestionTakenByOther(){
+        setUpReplyUseCase();
+        PostRequestModel inputData = new PostRequestModel(QUESTION_ID, ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData);
+
+        PostRequestModel inputData2 = new PostRequestModel(QUESTION_ID, SECOND_ATTORNEY_ID, "Test text");
+        postInputBoundary.createPost(inputData2);// AttorneyQuestionList is not updated
+
+        User user = attorneyGateway.getUser(ATTORNEY_ID);
+        Question attorneyquestion = user.getQuestionsList().get(0);
+        assertEquals(attorneyquestion.getQuestionId(), QUESTION_ID);
+        assertEquals(attorneyquestion.getTakenByAttorney(), ATTORNEY_ID);
+        assertEquals(attorneyquestion.isTaken(), true);
+    }
+
+}
```

Comment:
I recommend adding a tearDown class with anotation AfterAll that deletes all entities that were added during setup and testing, as allowing them to stay in the database may result in problems in the future as well as taking unnecessary space.